language/move-lang/src/parser/lexer.rs

```diff
@@ -252,8 +251,30 @@ fn find_token(file: &'static str, text: &str, start_offset: usize) -> Result<(To
             }
         }
         'A'..='Z' | 'a'..='z' | '_' => {
-            let len = get_name_len(&text);
-            (get_name_token(&text[..len]), len)
+            if text.starts_with("x\"") && text.len() > 2 {
```

Comment:
Nit: do we want to allow both `xABD` and `XABC` for consistency with the address syntax (which allows both upper and lowercase x)?

I'm happy with only lowercase too (more readable IM), but it seems like whatever we do should be consistent with the address syntax.

Reply:
We allow `0XABC`?

Reply:
I'm neutral on that one. I'll plan to add the uppercase "X" option in a follow-up in no one objects.

Reply:
I kind of don't like the X option. Should we really have it for the 0x numbers? Is it that useful/common/expected?

Reply:
C and C++ both allow the uppercase "0X" prefix. I don't think that Rust allows it, though. I don't like uppercase for the actual hex values, either, but I think it's good to allow flexibility on that one. I'd be fine to allow only lowercase "0x" if there is a consensus around that.

Reply:
I say less is better here. Just 0x seems fine by me

Reply:
Agreed; I actually prefer lowercase only. My only argument for the uppercase `X` was consistency.